geomesa-core/src/main/scala/org/locationtech/geomesa/core/transform/TransformCreator.scala

```diff
@@ -38,21 +38,11 @@ object TransformCreator {
 
     val encoder = SimpleFeatureEncoder(targetFeatureType, featureEncoding)
     val defs = TransformProcess.toDefinition(transformString)
-    featureEncoding match {
-      case FeatureEncoding.KRYO | FeatureEncoding.AVRO =>
-        (feature: SimpleFeature) => {
-          val newSf = new ScalaSimpleFeature(feature.getIdentifier.getID, targetFeatureType)
-          defs.foreach { t => newSf.setAttribute(t.name, t.expression.evaluate(feature)) }
-          encoder.encode(newSf)
-        }
 
-      case FeatureEncoding.TEXT =>
-        val builder = ScalaSimpleFeatureFactory.featureBuilder(targetFeatureType)
-        (feature: SimpleFeature) => {
-          defs.foreach { t => builder.set(t.name, t.expression.evaluate(feature)) }
-          val newFeature = builder.buildFeature(feature.getID)
-          encoder.encode(newFeature)
-        }
-    }
+    (feature: SimpleFeature) => {
+        val newSf = new ScalaSimpleFeature(feature.getIdentifier.getID, targetFeatureType)
```

Comment:
you could create the simple feature outside the function and re-use it. You can do newSf.getIdentifier.setID and set the attributes as you do.